plutus-contract/test/Spec/Emulator.hs

```diff
@@ -314,6 +315,13 @@ pubKeyTransactions2 = do
     Trace.liftWallet wallet1 $ payToPaymentPublicKeyHash_ def W.always (Ada.adaValueOf 20) pubKey2
     void Trace.nextSlot
 
+evalEmulatorTraceTest :: Property
+evalEmulatorTraceTest = property $ do
+    let trace = Trace.payToWallet wallet1 wallet2 (Ada.adaValueOf 10)
+        res = Trace.evalEmulatorTrace def trace
+    Hedgehog.annotateShow res
+    Hedgehog.assert (either (const False) (const True) res)
```

Comment:
I think we also need to test that the result returned by the trace is the correct one.

For example, something like

```
let trace = do
     Trace.payToWallet wallet1 wallet2 (Ada.adaValueOf 10)
     pure "DONE" -- Or something else
...
Hedgehog.assert (either (const False) (\x -> x == "DONE") res)
```

Reply:
The type ensures that, it can't pluck a `TxId` from thin air. And the actual value of the TxId can change so I don't wan't to test for that.

Reply:
I'm not sure I understand. I wasn't talking about `TxId`. I'm just thinking of returning a value from the trace and then verify that `evalEmulatorTrace` correctly returns that value.

Reply:
`Trace.payToWallet` returns an `EmulatorTrace TxId`. Which means `res` is of type `Either EmulatorErr TxId`. So if `res` is a `Right something` that something has to be the `TxId` returned by `payToWallet`.

Reply:
But doesn't that specific test ensure that you implemented the correct behavior? For example, if you returned the result of the wrong thread in `runThreads` (because `exit` was called by another thread), then you would get the wrong `Right something`, no?

Reply:
That could only happen is that other thread would call `exit` with a `TxId`, and that could only be the `TxId` created by `payToWallet`. Replacing the `TxId` with a `String` doesn't change anything.

Reply:
That's the power of type variables, theorems for free! https://www.cs.utah.edu/~mflatt/past-courses/cs7520/public_html/s06/wadler89.pdf